# Estatisticas Descrítivas
---
#### Estatísticas descritivas sob dataset ccmec25a64_main.csv

#### Importação, preparação e pré-organização de dados

In [ ]:
# Calculos, medias e manipulacao de dados
import pandas as pd
import math as mt
import numpy as np
import statistics as sta

# Geracao de Graficos
from matplotlib import pyplot as plt

# PATHs Variaveis de ambientes e etc
CSV_PATH = '../../Datasets/ccmec25a64_main.csv'

In [ ]:
# Carregamento do dataset
df = pd.read_csv(CSV_PATH)
df.head

# Limpeza basica de dados (Notebook1)
df['dt_competencia'] = pd.to_datetime(df['dt_competencia'])
df['dt_atualizacao'] = pd.to_datetime(df['dt_atualizacao'])

df = df.sort_values(by=['dt_competencia', 'sg_uf', 'no_municipio'])

qtd_zeros = len(df[df['vl_indicador_calculado_mun'] == 0])

print(f"Total de registros na base: {len(df)}")
print(f"Registros municipais com valor zero: {qtd_zeros} ({(qtd_zeros/len(df))*100:.2f}%)")


df_limpo = df[df['vl_indicador_calculado_mun'] > 0].copy()
print(f"Registros válidos (>0) para análise descritiva: {len(df_limpo)}")


Total de registros na base: 55702
Registros municipais com valor zero: 49716 (89.25%)
Registros válidos (>0) para análise descritiva: 5986


#### Medidas de Posição e Separatrizes
---
Calculo da Média, Mediana, Moda <br>
Quartis, Decis e Percentis

Estes calculos serão feitos sob as colunas 'vl_indicador_calculado_mun', 'vl_indicador_calculado_uf' e 'vl_indicador_calculado_br'


In [ ]:

def calcPosicaoESeparatrizes(dataframe_col, porcentagens = [0.10, 0.25, 0.50, 0.75, 0.90, 0.95]):
        
        return {
                'media': np.average(dataframe_col),
                'mediana': np.median(dataframe_col),
                'moda': sta.mode(dataframe_col),

                'percentrizes': dataframe_col.quantile(porcentagens)
        }
vl_indicador_calculado_mun = df_limpo['vl_indicador_calculado_mun']
vl_indicador_calculado_uf = df_limpo['vl_indicador_calculado_uf']
vl_indicador_calculado_br = df_limpo['vl_indicador_calculado_br']

# Municipio & Medidas de Posicao

municipio_dados = calcPosicaoESeparatrizes(vl_indicador_calculado_mun)
distrito_dados = calcPosicaoESeparatrizes(vl_indicador_calculado_uf)
brasil_dados = calcPosicaoESeparatrizes(vl_indicador_calculado_br)

print('Valores por municipio: \n ')

for chave, valor in municipio_dados.items():
        print(f'{chave}: {valor}')

print('Valores por distrito: \n ')
for chave, valor in distrito_dados.items():
        print(f'{chave}: {valor}')

print('Valores no brasil ao todo: ')
for chave, valor in brasil_dados.items():
        print(f'{chave}: {valor}')




Valores por municipio: 
 
Valores por distrito: 
 
media: 497471.0253925827
mediana: 295805.0
moda: 1924582.0
percentrizes: 0.10      98552.0
0.25     146417.0
0.50     295805.0
0.75     507748.0
0.90    1730813.0
0.95    1874035.0
Name: vl_indicador_calculado_uf, dtype: float64
Valores no brasil ao todo: 
media: 6791345.48346141
mediana: 6956725.0
moda: 6944755.0
percentrizes: 0.10    6425019.0
0.25    6805670.0
0.50    6956725.0
0.75    7292814.0
0.90    7915957.0
0.95    7915957.0
Name: vl_indicador_calculado_br, dtype: float64


#### Dados Avaliados

---

##### Valor indicador calc Por município

**Média Calculada por município**: 11317.16 </br>
**Mediana Calculada por município**: 4018.50 </br>
**Moda Calculada** (valor que mais se repete): 1.0 <br>

**Percentis, Quartis e Decis**

| Percentis | Valor    | Dado Eq      |
| --------- | -------- | ------------ |
| P10       | 319.00   | ---          |
| P25       | 1075.00  | Q1           |
| P50       | 4018.50  | Q2 - Mediana |
| P75       | 10601.25 | Q3           |
| P90       | 23027.50 | ---          |
| P95       | 38411.50 | ---          |

---

##### Valor indicador calc Por distrito

**Média Calculada por distrito**: 497471.03 </br>
**Mediana Calculada por distrito**: 295805.00 </br>
**Moda Calculada** (valor que mais se repete): 1924582.00 <br>

**Percentis, Quartis e Decis**

| Percentis | Valor      | Dado Eq      |
| --------- | ---------- | ------------ |
| P10       | 98552.00   | ---          |
| P25       | 146417.00  | Q1           |
| P50       | 295805.00  | Q2 - Mediana |
| P75       | 507748.00  | Q3           |
| P90       | 1730813.00 | ---          |
| P95       | 1874035.00 | ---          |

---

##### Valor indicador calc Brasil (total)

**Média Calculada Brasil**: 6791345.48 </br>
**Mediana Calculada Brasil**: 6956725.00 </br>
**Moda Calculada** (valor que mais se repete): 6944755.00 <br>

**Percentis, Quartis e Decis**

| Percentis | Valor      | Dado Eq      |
| --------- | ---------- | ------------ |
| P10       | 6425019.00 | ---          |
| P25       | 6805670.00 | Q1           |
| P50       | 6956725.00 | Q2 - Mediana |
| P75       | 7292814.00 | Q3           |
| P90       | 7915957.00 | ---          |
| P95       | 7915957.00 | ---          |

---

### Análise dos dados

---

#### Análise por município

**90% dos dados dessa coluna (P90)** são menores ou iguais a **23027.50**, ou seja, 90% dos municípios possuem um valor indicador de até esse limite. <br>
**95% dos dados (P95)** são menores ou iguais a **38411.50**. <br>
Os valores acima desse intervalo representam os casos mais extremos da distribuição, podendo indicar **outliers superiores**.

A diferença entre a **média (11317.16)** e a **mediana (4018.50)** pode sr uma **assimetria à direita**.

---

#### Análise por distrito

**90% dos distritos** possuem valor indicador menor ou igual a **1730813.00**. <br>
**95% dos distritos** apresentam valores de até **1874035.00**.

A média (**497471.03**) está acima da mediana (**295805.00**), sugerindo também **assimetria à direita**.

Há também uma diferença, **P75 (507748.00)** e **P90 (1730813.00)** mostra um crescimento acentuado nos maiores valores, indicando possível presença de **outliers**.

---

#### Análise Brasil

**90% dos valores nacionais** são menores ou iguais a **7915957.00**. <br>
**P95 apresenta o mesmo valor do P90**, baixa variação entre os dois registros.

A média (**6791345.48**) e a mediana (**6956725.00**) estão próximas, indicando uma distribuição **mais equilibrada** em comparação com município e distrito.


### Medidas de dispersão
---

Calculos de Amplitude, Variância, Desvio Padrão e Coeficiente de variação

Esses cálculos serão feitos sob as mesmas colunas listadas anteriormente.

In [ ]:
# Definição de colunas
# TODO: Padronizar uso de colunas no Notebook

cols = [
     "vl_indicador_calculado_mun",
    "vl_indicador_calculado_uf",
    "vl_indicador_calculado_br"
]

for col in cols:
    amplitude = 